types/boom/boom-tests.ts

```diff
@@ -136,17 +136,53 @@ interface CustomPayload extends Boom.Payload {
 (error.output.payload as CustomPayload).custom = 'abc_123';
 
 /**
- * Test additional data
+ * Test custom data
  */
 
-// If the concrete functions do not default their generic parameter to null, too, this fails:
-const error2: Boom.BoomError = Boom.badImplementation('really bad');
+// If the concrete functions do not default their generic parameter to null, this assignment fails:
+const error2: Boom.BoomError = Boom.badImplementation('message');
 
-// And this too:
-const next = <Hapi.ContinuationValueFunction>(() => { });
-next(Boom.badData());
+// If the interface defaults its generic parameter to null, you cannot pass it as a parameter to existing continuation functions:
+const handleError: Hapi.ContinuationValueFunction = (err?: Boom.BoomError | null) => {
```

Comment:
The hapi examples are very useful to have but they should be moved to the hapi test code (i.e. change this PR to `[boom, hapi] Add default generic type to all boom error creation functions`.